// react-pure-loaders/build/BallZigZagDeflect/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _core = require("@emotion/core");

var _mixins = require("../mixins");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var amount = '30px';
var ballZigDeflect = (0, _core.keyframes)({
  '100%': {
    transform: 'translate(0, 0)'
  },
  '17%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  },
  '34%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  },
  '50%': {
    transform: 'translate(0, 0)'
  },
  '67%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  },
  '84%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  }
});
var ballZagDeflect = (0, _core.keyframes)({
  '100%': {
    transform: 'translate(0, 0)'
  },
  '17%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  },
  '34%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  },
  '50%': {
    transform: 'translate(0, 0)'
  },
  '67%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  },
  '84%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  }
});

var _default = function _default(color) {
  return {
    position: 'relative',
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, -").concat(parseInt(amount, 10) / 2, "px)"),
    ' > div': _objectSpread({}, (0, _mixins.balls)(color), {}, (0, _mixins.globalAnimation)(), {
      left: '-7px',
      marginLeft: parseInt(amount, 10) / 2,
      position: 'absolute',
      top: '4px',
      '&:first-of-type': {
        animation: "".concat(ballZigDeflect, " 1.5s 0s infinite linear")
      },
      '&:last-child': {
        animation: "".concat(ballZagDeflect, " 1.5s 0s infinite linear")
      }
    })
  };
};

exports["default"] = _default;
