// react-pure-loaders/build/BallScaleRippleMultiple/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _core = require("@emotion/core");

var _mixins = require("../mixins");

var _utilities = _interopRequireDefault(require("../utilities"));

var _variables = require("../variables");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var size = '50px';
var ballScaleRippleMultiple = (0, _core.keyframes)({
  '0%': {
    opacity: 1,
    transform: 'scale(0.1)'
  },
  '100%': {
    opacity: 0.0
  },
  '70%': {
    opacity: 0.7,
    transform: 'scale(1)'
  }
});

function ballScaleRipple() {
  var n = 3;
  var i = 0;
  var grid = {};

  for (; i < 4; i += 1) {
    grid["> div:nth-of-type(".concat(i, ")")] = {
      animationDelay: (0, _utilities["default"])('0.2s', n, i - 1)
    };
  }

  return grid;
}

var _default = function _default(color) {
  return _objectSpread({}, ballScaleRipple(), {
    position: 'relative',
    transform: "translateY(-".concat(parseInt(size, 10) / 2, ")"),
    ' > div': _objectSpread({}, (0, _mixins.globalAnimation)(), {
      animation: "".concat(ballScaleRippleMultiple, " 1.25s 0s infinite cubic-bezier(.21,.53,.56,.8)"),
      border: "2px solid ".concat(color || _variables.PRIMARY_COLOR),
      borderRadius: '100%',
      height: size,
      left: '-26px',
      position: 'absolute',
      top: '-2px',
      width: size
    })
  });
};

exports["default"] = _default;
