// react-pure-loaders/build/esm/BallZigZagDeflect/style.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import { keyframes } from '@emotion/core';
import { balls, globalAnimation } from '../mixins';
var amount = '30px';
var ballZigDeflect = keyframes({
  '100%': {
    transform: 'translate(0, 0)'
  },
  '17%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  },
  '34%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  },
  '50%': {
    transform: 'translate(0, 0)'
  },
  '67%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  },
  '84%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, -").concat(amount, ")")
  }
});
var ballZagDeflect = keyframes({
  '100%': {
    transform: 'translate(0, 0)'
  },
  '17%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  },
  '34%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  },
  '50%': {
    transform: 'translate(0, 0)'
  },
  '67%': {
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  },
  '84%': {
    transform: "translate(".concat(parseInt(amount, 10) / 2, "px, ").concat(amount, ")")
  }
});
export default (function (color) {
  return {
    position: 'relative',
    transform: "translate(-".concat(parseInt(amount, 10) / 2, "px, -").concat(parseInt(amount, 10) / 2, "px)"),
    ' > div': _objectSpread({}, balls(color), {}, globalAnimation(), {
      left: '-7px',
      marginLeft: parseInt(amount, 10) / 2,
      position: 'absolute',
      top: '4px',
      '&:first-of-type': {
        animation: "".concat(ballZigDeflect, " 1.5s 0s infinite linear")
      },
      '&:last-child': {
        animation: "".concat(ballZagDeflect, " 1.5s 0s infinite linear")
      }
    })
  };
});
