// react-pure-loaders/build/BallScaleMultiple/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ballScale = ballScale;
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _core = require("@emotion/core");

var _mixins = require("../mixins");

var _utilities = _interopRequireDefault(require("../utilities"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var size = '60px';
var ballScaleMultiple = (0, _core.keyframes)({
  '0%': {
    opacity: 0,
    transform: 'scale(0.0)'
  },
  '100%': {
    opacity: 0,
    transform: 'scale(1.0)'
  },
  '5%': {
    opacity: 1
  }
});

function ballScale() {
  var n = 3;
  var i = 2;
  var grid = {};

  for (; i < 4; i += 1) {
    grid["> div:nth-of-type(".concat(i, ")")] = {
      animationDelay: (0, _utilities["default"])('0.2s', n, i)
    };
  }

  return grid;
}

var _default = function _default(color) {
  return _objectSpread({}, ballScale(), {
    position: 'relative',
    transform: "translateY(-".concat(size, " / 2)"),
    ' > div': _objectSpread({}, (0, _mixins.balls)(color), {}, (0, _mixins.globalAnimation)(), {
      animation: "".concat(ballScaleMultiple, " 1s 0s linear infinite"),
      height: size,
      left: '-30px',
      margin: '0',
      opacity: 0,
      position: 'absolute',
      top: '0px',
      width: size
    })
  });
};

exports["default"] = _default;
