// react-pure-loaders/build/Pacman/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _core = require("@emotion/core");

var _mixins = require("../mixins");

var _utilities = _interopRequireDefault(require("../utilities"));

var _variables = require("../variables");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var size = '25px';
var rotatePacmanHalfUp = (0, _core.keyframes)({
  '0%': {
    transform: 'rotate(270deg)'
  },
  '100%': {
    transform: 'rotate(270deg)'
  },
  '50%': {
    transform: 'rotate(360deg)'
  }
});
var rotatePacmanHalfDown = (0, _core.keyframes)({
  '0%': {
    transform: 'rotate(90deg)'
  },
  '100%': {
    transform: 'rotate(90deg)'
  },
  '50%': {
    transform: 'rotate(0deg)'
  }
});
var pacmanBalls = (0, _core.keyframes)({
  '100%': {
    transform: "translate(-".concat(4 * parseInt(size, 10), "px, -").concat(parseInt(size, 10) / 4, "px)")
  },
  '75%': {
    opacity: 0.7
  }
});

var pacmanDesign = function pacmanDesign(color) {
  return {
    borderBottom: "".concat(size, " solid ").concat(color || _variables.PRIMARY_COLOR),
    borderLeft: "".concat(size, " solid ").concat(color || _variables.PRIMARY_COLOR),
    borderRadius: size,
    borderRight: "".concat(size, " solid transparent"),
    borderTop: "".concat(size, " solid ").concat(color || _variables.PRIMARY_COLOR),
    height: '0px',
    width: '0px'
  };
};

function ballPlacement() {
  var n = 3;
  var i = 0;
  var grid = {};

  for (; i < 4; i += 1) {
    grid["> div:nth-of-type(".concat(i + 2, ")")] = {
      animation: "".concat(pacmanBalls, " 1s ").concat((0, _utilities["default"])('.33s', n, i), " infinite linear")
    };
  }

  return grid;
}

var _default = function _default(color) {
  return _objectSpread({}, ballPlacement(), {
    position: 'relative',
    ' > div:first-of-type': _objectSpread({}, pacmanDesign(color), {
      animation: "".concat(rotatePacmanHalfUp, " 0.5s 0s infinite"),
      left: '-30px',
      position: 'relative'
    }),
    ' > div:nth-of-type(2)': _objectSpread({}, pacmanDesign(color), {
      animation: "".concat(rotatePacmanHalfDown, " 0.5s 0s infinite"),
      left: '-30px',
      marginTop: -2 * parseInt(size, 10),
      position: 'relative'
    }),
    ' > div:nth-of-type(3), > div:nth-of-type(4), > div:nth-of-type(5), > div:nth-of-type(6)': _objectSpread({}, (0, _mixins.balls)(color), {
      height: '10px',
      left: '70px',
      position: 'absolute',
      top: '25px',
      transform: "translate(0,-".concat(parseInt(size, 10) / 4, "px)"),
      width: '10px'
    })
  });
};

exports["default"] = _default;
